R/otel.R

```diff
@@ -453,7 +474,7 @@ create_ospan_promise_domain <- function() {
       # the inconsistent behavior.
 
       span <- get_active_span()
-      if (!span$is_recording()) {
+      if (!.subset2(span, "is_recording")()) {
```

Comment:
This method is not a "safe" method :-/. We should use a different check.

Maybe inherits `"otel_span_noop"` or `tracer$is_enabled()`?

(If `tracer` is used, should we require it be passed in? Shiny could be updated to do so. Feels like the internal promises tracer could be init'ed just before `new_promise_domain()`.)